scripts/devaccess/devaccess_wrap.py

```diff
@@ -0,0 +1,455 @@
+#!/usr/bin/python
+# vim: expandtab:tabstop=4:shiftwidth=4
+'''
+    Tool to provide developers with ability to remotely run a limited
+    set of commands.
+'''
+# Disabling invalid-name because pylint doesn't like the naming conention we have.
+# pylint: disable=invalid-name
+
+import logging
+import os
+import re
+import subprocess
+import sys
+import yaml
+
+class OCCmd(object):
+    ''' Class to hold and standardize building of /usr/bin/oc commands
+        out of raw string command line text '''
+    def __init__(self, raw_cmd):
+        # original command line text
+        self._raw_cmd = raw_cmd
+
+        self._params = {}
+        # always define a namespace
+        self._params['namespace'] = 'default'
+        # whether special output formatting is requested
+        self._params['output_format'] = None
+        # --follow or -f
+        self._params['follow'] = None
+        # -c for 'oc get logs -c <container>
+        self._params['container'] = None
+
+        # the actual command to run without the params
+        # (ie oc get pods or oc get secrets my-secret)
+        self._params['verb'] = None
+        self._params['type'] = None
+        self._params['subject'] = None
+        self.parse_cmd()
+
+    def get_namespace(self, cmd):
+        ''' find a namespace (if passed in) and return cmd
+            without the namespace-related parameters '''
+
+        cmd_split = cmd.split()
+        delete_tokens = []
+
+        for x in range(0, len(cmd_split)):
+            if cmd_split[x] == '-n':
+                # token was exactly '-n' so next token is the namespace
+                self._params['namespace'] = cmd_split[x+1]
+
+                # make sure to mark these items for removal from list
+                # before returning the command
+                delete_tokens.append(cmd_split[x])
+                delete_tokens.append(cmd_split[x+1])
```

Comment:
This doesn't look safe, and will behave oddly if tokens are not unique.
For consideration: "command -a foo -b foo".  If this code fragment were used here, it would delete both occurrences of "foo".
Can you use "import argparse" for processing instead?